vmangos/core
# Core/Spell - move corpse check for cannibalize to Spell::CheckCast

## 🍰 Pullrequest
Improvements for spell https://classic.wowhead.com/spell=20577

- move cannibalize corpse check into `SpellCastResult Spell::CheckCast(bool strict)` like cmangos and remove cooldown hack

- ~~prevent cannibalize sending interrupted message, when cast is succesful (I have no idea how to fix this appart from checking for the spell id, any suggestions are welcome)~~ since checking for the spell id was not the fix for the actual problem, I've removed that change

### Proof
https://github.com/cmangos/mangos-classic/blob/5f8b6afb732237f8973d2c475eb10d8eae4839fc/src/game/Spells/Spell.cpp#L4877

https://github.com/cmangos/mangos-classic/blob/3b534fd702aac54be554a0465d7dcda05665c9ce/src/game/Spells/SpellEffects.cpp#L845

### Issues
- relates https://github.com/vmangos/core/issues/1334

### How2Test
- create undead character
- use cannibalize on a valid dead corpse
- ~~cannibalize is casted -> player should not see interrupted message~~
- cannibalize should work like before
